ksqldb-common/src/main/java/io/confluent/ksql/util/ConsistencyOffsetVector.java

```diff
@@ -222,16 +176,8 @@ public boolean equals(final Object o) {
     if (o == null || getClass() != o.getClass()) {
       return false;
     }
-    final ConsistencyOffsetVector other = (ConsistencyOffsetVector) o;
-
-    try {
-      rwLock.readLock().lock();
-      other.rwLock.readLock().lock();
-      return this.version == other.version && offsetVector.equals(other.offsetVector);
-    } finally {
-      rwLock.readLock().unlock();
-      other.rwLock.readLock().unlock();
-    }
+    final ConsistencyOffsetVector vector1 = (ConsistencyOffsetVector) o;
```

Comment:
Given we have decided to keep the client exposed type ConsistencyOffsetVector separate from the internal Position type, shouldn't we then make sure we don't expose Position in any of those methods? For instance, we can either make this logic above a non-public translation method, or we do it outside this class.

Reply:
I am not sure I follow. Where is the Position exposed to the client? The code you are pointing is the `equals`. What is wrong here?

Reply:
I think @patrickstuedi was referring to `public void updateFromPosition(Position position)` above. GH doesn't let you comment on arbitrary lines in a PR, so sometimes you have to just comment on a random line.

Philosophically, I agree. If we want to achieve good decoupling, the way to do it is not reference Position or ConsistencyOffsetVector at all from each other, but have a separate utility method to translate between them.

On the other hand, ConsistencyOffsetVector is not a public API, and it is itself a utility (it's in the `util` package), so that method isn't too much of a sin.

In the interest of forward progress, I'd be happy to accept this PR as-is, but I'd be equally happy to see Patrick's suggestion adopted.